### PR TITLE
Adds autocomplete for coffeescript

### PR DESCRIPTION
**Pull request by sdwolf**
Refs #5896 

This is a naive take on activating company-mode for CoffeeScript.

Reviews welcome!

### CODE REVIEW

**Comment by JuanCaicedo** [comment]
This works great for me! I'll leave the reviewing to someone with more experience with the source code :)

File: layers/+lang/javascript/packages.el [hunk 1]
```diff
@@ -37,7 +38,8 @@
       (add-hook 'coffee-mode-hook
                 '(lambda ()
                    (setq indent-line-function 'javascript/coffee-indent
-                         evil-shift-width coffee-tab-width))))))
+                         evil-shift-width coffee-tab-width)
+                   (company-mode 1))))))
```
Comment (TheBB) [inline layers/+lang/javascript/packages.el]: Please rewrite to use `spacemacs|add-company-backends`.

**Comment by sdwolf** [comment]
I've used `company-capf` since coffeescript does not have a dedicated backend (inspired from the `idris` layer). 

The `company` package is already in the `javascript-packages` list, is there any other place I need to add it as a dependency?

**Comment by TheBB** [comment]
If you've added the package to the package list, you need to do the initialization that depends on that package in the `javascript/post-init-company` function. Not in `javascript/init-coffee-mode`. Otherwise people who use the javascript layer but not autocompletion will find themselves confused as to why they're getting errors about no function being called `company-mode`.

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

**Pull request closed without merging**